egs/librispeech/ASR/pruned_transducer_stateless2/beam_search.py

```diff
@@ -600,7 +616,36 @@ def greedy_search_batch(
         # logits'shape (batch_size, 1, 1, vocab_size)
 
         logits = logits.squeeze(1).squeeze(1)  # (batch_size, vocab_size)
-        assert logits.ndim == 2, logits.shape
+
+        if ngram_rescoring:
+            all_logits[start:end] = logits
+
+            assert logits.ndim == 2, logits.shape
+            logits_argmax = logits.argmax(dim=1)
+            logits_softmax = logits.softmax(dim=1)
+
+            # detailed in below fuction verify_non_blank_logits.
+            selection_verification = True
+
+            # 0 for blank frame and 1 for non-blank frame.
+            non_blank_flag[start:end] = torch.where(
+                logits_argmax == blank_id, 0, 1
+            )
+
+            if False:
+                # In paper: https://arxiv.org/pdf/2101.06856.pdf
+                # A gama_blank threshold value is used to determinze blank frame.
+                # Currently, results are worse than baseline greedy_search
+                #  and also very sensitive to gama_blank.
+                # (TODO): debug this later.
+                gama_blank = 0.50
```

Comment:
Did you try large gama_blank value? 0.5 means p_t_u(blank) > 0.5 would be skipped. Maybe 0.95 or 0.99 better?

Reply:
Thanks for the information.
Actually I tried several weights and 0.5 is the best among them.  Also the result is quite sensitive to game_blank.
Maybe  because we use different WFSTDecoding methods, or current 3-gram is not appropriate for this task.
gama_blank|test-clean|detail error
--|--|--
0.4|20.75|Errors: 365 insertions, 5187 deletions, 5357 substitutions, over 52576 reference words
0.5|2.57|Errors: 331 insertions, 81 deletions, 994 substitutions, over 52576 reference words
0.6|21.66|Errors: 7788 insertions, 146 deletions, 3456 substitutions, over 52576 reference words